acceptance/tests/agent/agent_parses_json_catalog.rb

```diff
@@ -0,0 +1,31 @@
+test_name "C99978: Agent parses a JSON catalog"
+tag 'risk:medium'
+
+require 'json'
+
+step "Refresh the catalog" do
+  on(agents, puppet("agent --test --server #{master}"))
+end
+
+step "Agent parses a JSON catalog" do
+  agents.each do |agent|
+    # Is it there?
+    json_catalog = File.join(agent.puppet['client_datadir'], 'catalog',
+                             "#{agent.puppet['certname']}.json")
+    on(agent, "[[ -f #{json_catalog} ]]", {:acceptable_exit_codes => [0]})
+
+    # Can we read it?
+    rc = on(agent, "cat #{json_catalog}", {:acceptable_exit_codes => [0]})
```

Comment:
This is going to dump the entire catalog to the beaker log, right? Should we redirect to `/dev/null` to keep the log cleaner, or is showing it to the user intentional? If intentional, can we add a comment to that effect?